### PR TITLE
Create endpoint for sending bulk emails

### LINKED ISSUE #318: Implement send email admin endpoint
**Definition of done**
There should be an admin endpoint that can take an object with recipients, subject, and content and send out the an email. 

**Things to consider**
- There should be a mechanism to get a predefined group (selected mentors, selected mentees, etc.) from the frontend, query for those email addresses and send out the email 
- There should be a mechanism to get a single email from the frontend and send the email to that 

More info: 
https://sef.discourse.group/t/engineering-team-weekly-stand-ups-2022/397/3?u=gravewalker

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #318 

## Goals

Providing an admin endpoint to send bulk emails for a specific group in a program or a set of custom emails

## Approach

- Created an Enum for mail group types (ALL, ALL_MENTORS, ALL_MENTEES, SELECTED_MENTORS, REJECTED_MENTORS, SELECTED_MENTEES, REJECTED_MENTEES)
- Created the model `BulkEmailDto` to use as the endpoint request body
- Created the `ProgramSevice.java:sendBulkEmails` method which gets the list of emails according to the group and sends the mails using EmailService. 
- Created the endpoint `POST /admin/program/{id}/email`
```
 {
   subject: String,
   message: String,
   mailGroup: Enum("ALL", "ALL_MENTORS", "ALL_MENTEES", "SELECTED_MENTORS", "REJECTED_MENTORS", "SELECTED_MENTEES", "REJECTED_MENTEES"),
       additionalEmails: [array of string] (to send the emails in them)
   },
 }
```

Used the programId as a path param since bulk emailing is only used with a single program.

### Screenshots

N/A

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
 
N/A

## Test environment
 
N/A

## Learning

N/A

### CODE REVIEW

File: src/main/java/org/sefglobal/scholarx/model/BulkEmailDto.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@Data
+public class BulkEmailDto {
+	private String subject;
+	private String message;
+	private MailGroup mailGroup;
+	private List<String> additionalEmails;
+
+	public String getSubject() {
+		return subject;
+	}
```
Comment (jayasanka-sack) [inline src/main/java/org/sefglobal/scholarx/model/BulkEmailDto.java:17]: We don't need to write getters and setters because you have imported lamblok

**Comment by Piumal1999** [comment]
There are several Enrolment states available for mentees and mentors other than Approved and Rejected. Previously we discussed only those 2 enrolment states, so there’s an issue with that approach.

We can’t keep a mail group for Rejected Mentees because the REJECTED state is used for identifying the mentees who are **rejected by a mentor**, and it doesn’t mean that the mentee is rejected from the program.

So I suggest these new mail groups for this mailing feature.

```
ALL: All mentors and mentees in the program
ALL_MENTORS: All mentors in the program
ALL_MENTEES: All mentees in the program

APPROVED_MENTORS: Mentors who are approved for the program
REJECTED_MENTORS: Mentors who are rejected from the program

APPROVED_MENTEES: Mentees who are approved for the program
NOT_SELECTED_MENTEES: Mentees who are not selected for the program. They will not be APPROVED for the program in a future state (They are either discarded or failed from wildcard)
```

File: src/main/java/org/sefglobal/scholarx/model/BulkEmailDto.java (hunk 1 above)
Comment (Piumal1999) [reply src/main/java/org/sefglobal/scholarx/model/BulkEmailDto.java:15]: Removed lombok

**Comment by Gravewalker666** [comment]
> NOT_SELECTED_MENTEES - They will not be APPROVED for the program in a future state (They are either discarded or failed from wildcard)

This implies that its all mentees but the already approved ones lets rename it to `DISCARDED_MENTEES`

File: src/main/java/org/sefglobal/scholarx/util/MailGroup.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.sefglobal.scholarx.util;
+
+public enum MailGroup {
+	ALL,
+	ALL_MENTORS,
+	ALL_MENTEES,
+	APPROVED_MENTORS,
+	REJECTED_MENTORS,
+	APPROVED_MENTEES,
+	NOT_SELECTED_MENTEES,
```
Comment (Gravewalker666) [inline src/main/java/org/sefglobal/scholarx/util/MailGroup.java:10]: Let's rename this to what I mentioned on the above comment

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]

**Comment by Gravewalker666** [comment]
@Piumal1999 there are some conflicts after merging anjula's PR can you resolve them

**Comment by Piumal1999** [comment]
> @Piumal1999 there are some conflicts after merging anjula's PR can you resolve them

Yeah. Resolved those with 02c325a

**Review by Gravewalker666 — APPROVED** [review]
Nice work, Thank you @Piumal1999

**Pull request merged**